test/mobile/lightweight_dispatch/tests_setup.py

```diff
@@ -1,203 +1,143 @@
-    path = 'divide_Tensor.ptl'
-
-    def setup(self):
-        class Model(torch.nn.Module):
-            def forward(self, b):
-                a = torch.tensor(3, dtype=torch.int64)
-                out = torch.empty(size=[1], dtype=torch.float)
-                torch.div(b, a, out=out)
-                return [torch.div(b, a, rounding_mode='trunc'), out]
-        model = Model()
-        # Script the model and save
-        script_model = torch.jit.script(model)
-        script_model._save_for_lite_interpreter(self.path)
-
-
-class ModelWithMultipleOps(FileSetup):
-    path = 'multiple_ops.ptl'
-
-    def setup(self):
-        class Model(torch.nn.Module):
-            def __init__(self):
-                super(Model, self).__init__()
-                self.ops = torch.nn.Sequential(
-                    torch.nn.ReLU(),
-                    torch.nn.Flatten(),
-                )
-
-            def forward(self, x):
-                x[1] = -2
-                return self.ops(x)
-
-        model = Model()
-        # Script the model and save
-        script_model = torch.jit.script(model)
-        script_model._save_for_lite_interpreter(self.path)
-
-
-tests = [
-    ModelWithDTypeDeviceLayoutPinMemory(),
-    ModelWithTensorOptional(),
-    ModelWithScalarList(),
-    ModelWithFloatList(),
-    ModelWithListOfOptionalTensors(),
-    ModelWithArrayOfInt(),
-    ModelWithTensors(),
-    ModelWithStringOptional(),
-    ModelWithMultipleOps(),
-]
-
-
-def setup():
-    for test in tests:
-        test.setup()
-
-
-def shutdown():
-    for test in tests:
-        test.shutdown()
+@save_model
+class ModelWithTensors(torch.nn.Module):
+    def forward(self, a):
+        b = torch.ones_like(a)
+        return a + b
+
+@save_model
+class ModelWithStringOptional(torch.nn.Module):
+    def forward(self, b):
+        a = torch.tensor(3, dtype=torch.int64)
+        out = torch.empty(size=[1], dtype=torch.float)
+        torch.div(b, a, out=out)
+        return [torch.div(b, a, rounding_mode='trunc'), out]
+
+
+@save_model
+class ModelWithMultipleOps(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.ops = torch.nn.Sequential(
+            torch.nn.ReLU(),
+            torch.nn.Flatten(),
+        )
+
+    def forward(self, x):
+        x[1] = -2
+        return self.ops(x)
 
 
 if __name__ == "__main__":
     command = sys.argv[1]
+    ops_yaml = sys.argv[2]
+    backup = ops_yaml + ".bak"
     if command == "setup":
-        setup()
+        tests = [
+            ModelWithDTypeDeviceLayoutPinMemory(),
+            ModelWithTensorOptional(),
+            ModelWithScalarList(),
+            ModelWithFloatList(),
+            ModelWithListOfOptionalTensors(),
+            ModelWithArrayOfInt(),
+            ModelWithTensors(),
+            ModelWithStringOptional(),
+            ModelWithMultipleOps(),
+        ]
+        shutil.copyfile(ops_yaml, backup)
+        with open(ops_yaml, 'a') as f:
```

Comment:
don't you need to prepend a \n? or hardcode that in the .yaml file?

Reply:
[lightweight_dispatch_ops.yaml](https://github.com/pytorch/pytorch/pull/80791/files#diff-2e57f80e10ce026e5af291cd7f4493a621a6ba5017b0ae619ebfa4dbfc763fbf) actually has a trailing newline.